[HAO ZEYU] iP

PROGRAMMERHAO opened the pull request:
### **Duke Pro**
>“Your mind is for having ideas, not holding them.” – David Allen

DukePro frees your mind of having to remember things you need to do. It's,

- **text-based**
- **easy to learn**
- ~~FAST~~ **SUPER FAST to use**

All you need to do is, 

 1. download it from [here](https://github.com/PROGRAMMERHAO/ip/releases/tag/v0.1).
 2. double-click it.
 3. add your tasks.
 4. let it manage your task for you 😉

And it is **FREE**!

Features:

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

PROGRAMMERHAO committed:
Level 1

PROGRAMMERHAO committed:
Level 2

PROGRAMMERHAO committed:
Level3

PROGRAMMERHAO committed:
Level4

PROGRAMMERHAO committed:
A-TextUITesting

PROGRAMMERHAO committed:
Level5

PROGRAMMERHAO committed:
Level6

PROGRAMMERHAO committed:
A-ENUM

PROGRAMMERHAO committed:
Level-6 Refinements

PROGRAMMERHAO committed:
Level7

PROGRAMMERHAO committed:
Level7

PROGRAMMERHAO committed:
Level8

PROGRAMMERHAO committed:
Merge branch 'branch-Level-8'

PROGRAMMERHAO committed:
Level8

PROGRAMMERHAO committed:
Merge branch 'branch-Level-8'

PROGRAMMERHAO committed:
A-MoreOOP

PROGRAMMERHAO committed:
A-Packages

PROGRAMMERHAO committed:
Merge remote-tracking branch 'origin/add-gradle-support'

PROGRAMMERHAO committed:
A-Gradle

PROGRAMMERHAO committed:
A-JUnit

PROGRAMMERHAO committed:
A-JavaDoc

PROGRAMMERHAO committed:
A-CodingStandard

PROGRAMMERHAO committed:
Level-9

PROGRAMMERHAO committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/ui/Ui.java

PROGRAMMERHAO committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/MarkDoneCommand.java
#	src/main/java/duke/command/MarkUndoneCommand.java
#	src/main/java/duke/command/ShowListCommand.java
#	src/main/java/duke/storage/Storage.java
#	src/main/java/duke/tasks/Deadline.java
#	src/main/java/duke/tasks/Event.java
#	src/main/java/duke/tasks/Task.java

PROGRAMMERHAO committed:
Refinements to Style

ruiyigan commented on data/duke.txt:
```diff
@@ -0,0 +1,4 @@
+E | 1 | say hi | from 2 to 5
```
Just a suggestion. You may want to consider adding duke.txt into your .gitignore file to avoid the constant file change when committing changes.

ruiyigan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,75 @@
+
+
+
+
```
Perhaps you can consider adding a comment for the Duke class briefly explaining what the Duke class do.

ruiyigan commented on src/main/java/duke/Commands.java:
```diff
@@ -0,0 +1,5 @@
+package duke;
+
+public enum Commands {
```
I like how you have an enum to manage the different commands.

ruiyigan commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,92 @@
+
+
+
+
```
Similar comments as above regarding the comment for the class.

ruiyigan commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,30 @@
+import duke.ui.Ui;
+
+
+
```
Similar comments as above regarding the comment on class (for all classes under command folder)

ruiyigan commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,99 @@
+package duke.tasks;
+
+public class Task {
```
Similar comments as above regarding comments on class.

ruiyigan commented on src/test/java/duke/tasks/EventTest.java:
```diff
@@ -0,0 +1,15 @@
+        assertEquals("[E][ ] finish homework (from: 2pm to: 4pm)", new Event("finish homework", "from 2pm to 4pm").toString());
+    }
+
+}
```
I think you can include a new line here. I don't think its compulsory but github warns you about it. I found a reason online in this link [here](https://www.quora.com/Why-is-it-a-common-practice-to-put-a-new-line-at-the-end-of-a-source-code-file) if you're interested.

ruiyigan reviewed (commented):
Overall, great job! 👍 Some minor issues I found

- Missing comments for classes (I may have missed a class or two) but I think it will be helpful to include briefly what the class function is.
- Perhaps not committing your txt file to github.

Other than that, the code looks good!
Please let me know if you have any queries regarding my comments.

PROGRAMMERHAO committed:
Level-10

PROGRAMMERHAO committed:
ignore unnecessary files

PROGRAMMERHAO committed:
Merge branch 'branch-level-10'

Flash161203 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,75 @@
+     * @param filePath The string representation of the file path
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DirectoryNotFoundException e) {
+            ui.showError(e.getMessage());
+            tasks = new TaskList();
+        } catch (FileNotFoundException e) {
+            ui.showError(e.getMessage());
+            tasks = new TaskList();
```
The naming used here is very straightforward and easy to understand 👍

Flash161203 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,135 @@
+        try {
+            Scanner s = new Scanner(file); // create a Scanner using the File as the source
+            while (s.hasNext()) {
+                String cur = s.nextLine();
```
Perhaps could have avoided the short form and have used a slightly more descriptive variable name (e.g. `currentLine`)?

I also noticed similar instances in several other places as well...

Flash161203 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,135 @@
+     * @return The task deleted
+     */
+    public Task delete(int index) {
+        Task removed = taskList.remove(index);
```
I feel that it might be better to use the name `removedTask` instead :)
I would sat the same about instances in several other places as well, (e.g. `markedTask`, `unmarkedTask`, etc.)

Flash161203 commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,45 @@
+     * @param storage The storage object
+     */
+    public void execute(TaskList list, Ui ui, Storage storage) {
+        ArrayList<Integer> found = new ArrayList<>();
```
Maybe a more descriptive name like `tasksFound`?

Flash161203 reviewed (commented):
Overall, I found the 'Naming' aspect of Code Quality to be well-satisfied. Just a few minor things to 'fix'. Great job! 😄

PROGRAMMERHAO committed:
Minor changes

PROGRAMMERHAO committed:
A-Assertions

PROGRAMMERHAO committed:
Merge pull request #2 from PROGRAMMERHAO/A-Assertions

A-Assertions

PROGRAMMERHAO committed:
Code Quality: refactor the code to improve the code quality

Currently some imports and functions are redundant and some places need
more comments for other developers to understand

Let's remove the redundant code and add more comments to make the code
clearer and more succint

Other developers will not msiunderstand the code and they will be able
to know why some parsing is done in that way

PROGRAMMERHAO committed:
Merge pull request #3 from PROGRAMMERHAO/heads/A-CodeQuality

Code Quality: refactor the code to improve the code quality

PROGRAMMERHAO committed:
C-Undo: enable the user to undo four types of commands

If a user makes a mistake he or she cannot undo it

Allowing user to undo give them the chance to recover from their mistake

Let's enable undo options for delete task, mark a task as done, mark a
task as undone, and add a task

Undoing an undo is not allowed because this user will be unlikely to do
so, and undoing show task and find task commands are trivial so they
cannot be undone as well.

PROGRAMMERHAO committed:
Corrected some errors in assert

PROGRAMMERHAO committed:
Corrected some assertion errors and the file creation procedure.

PROGRAMMERHAO committed:
Refactored some code to SLAP

PROGRAMMERHAO committed:
Update User guide

Updated the user guide.

PROGRAMMERHAO committed:
Update User Guide

Update user guide

PROGRAMMERHAO committed:
Uploaded Product Screenshot

Uploaded Product Screenshot

PROGRAMMERHAO committed:
Delete Ui.png

PROGRAMMERHAO committed:
Uploaded Ui.png

Uploaded Ui.png

PROGRAMMERHAO committed:
Refactored code to align with code quality

PROGRAMMERHAO committed:
Merge branch 'master' of https://github.com/PROGRAMMERHAO/ip

PROGRAMMERHAO committed:
Improved Coding Style

PROGRAMMERHAO committed:
Delete Ui.png

PROGRAMMERHAO committed:
Added Ui.png

PROGRAMMERHAO committed:
Added more error handling functionalities to DUuke

PROGRAMMERHAO committed:
Merge branch 'master' of https://github.com/PROGRAMMERHAO/ip

PROGRAMMERHAO committed:
Update README.md
